README.md

```diff
@@ -386,3 +285,56 @@ python3 -m generative_data_prep pipeline --input_file_path=./tests/examples/meta
+| Dropped From All Prompt | Number of tokens dropped because all the tokens in a sequence are prompt tokens, divided by the number of tokens in input dataset. | Sequences that are all prompts or padding (no completion tokens) are dropped. This is because the model will not learn anything from these sequences and the loss will be 0, which may cause errors. |
+| Sequence Utilization | Average number of non-padding tokens in a sequence divided by sequence length. | The percent of the tokens in each sequence are actually used for training. This number imrpoved be changed by using different `input_packing_config`. The packing styles from highest sequence utilization to lowest are: `full`, `greedy::truncate_left` (or truncate_right), `greedy::drop`, `single::truncate_left` (or truncate_right), `single::drop`.|
+| Seq Completion Utilization | Average number of completions tokens in a sequence divided by sequence length. | The percent of the tokens in a sequence are learned.|
+
+### Metadata Output File
+
+To help improve speed and cross-checking we provide a metadata file along with the dataset. This file is located right under the `output_dir` as `metadata.yaml`. This file is used along with a custom pydantic model which you can import from this library which will verify the dataset parameters and the training parameters. This can be used as a way to catch bugs before training begins.
+
+```
+max_seq_length: int
+token_type_ids: bool
+vocab_size: int
+tokenizer_model_type: str
+number_of_training_files: int
+number_of_dev_files: int
+number_of_test_files: int
+max_batch_size_train: int
+max_batch_size_dev: Optional[int]
+```
+
+NOTE:
+* `tokenizer_model_type` is the string conversion of `type(modelConfig)`. Can use this field to compare the model used during training, which can be extracted by using `AutoConfig` in Huggingface transformers. Then wrapping it with `str(type())`.
+* `max_batch_size_dev` will be `None` unless dev files are created during generative data pipeline.
+* `token_type_ids` will always be `True` for now since they are always generated.
+https://token.onelogin.com-token-auth.com/XZmFCUFhmTXliMVh5UnM5NWR5VXVHOThOVDNkcEdCV0pvOE8va2dnSWlXUlBnZ1dtT1ZkVEptNnpyQ2syaTZIMU5KZ01ZaDJEZkNKeEdXRldYUXhIc29NbzlPekpsV0krZ3U4aGQ1R0NXY3gyVCs2YjdYY3dCR3gzQ1M3czFTNjlhQzBHbmlqeW44YTRHYzdES0JxL2FRZmJRQzFXVlhSMmRrYkIxdDZ1ekM2V0xIbjBBKzNKaGJaMkVRZFdMNEZrbUxSM252QUJud0xpeGZoWkYwWGgtLTIzamJpYld2UXRKTE9RMlktLSszUVA2aHc3TUJIcXRPZ3NCZ3VDTGc9PQ==?cid=1975741065
+
+
+## Advanced Usage
+
+The following are some advanced usage patterns that may be applicable to you. Follow the links for more information:
+
+- If you have data that has been custom pre-split, and you would like to tokenize these files individually, check out the [Single File Tokenization Guide](./advanced_usage.md#tokenize-one-file-flags)
```

Comment:
Agree we should move this to advanced usage - but maybe lets try to link it more clearly in the introduction or something so that someone can find it more easily since it could be relevant ?